gpytorch/lazy/cached_cg_lazy_tensor.py

```diff
@@ -134,19 +134,21 @@ def requires_grad(self):
     def requires_grad(self, val):
         self.base_lazy_tensor.requires_grad = val
 
-    def _cholesky(self):
+    def _cholesky(self, upper=False):
+        from .triangular_lazy_tensor import TriangularLazyTensor
+
         res = self.__class__(
-            self.base_lazy_tensor._cholesky(),
+            self.base_lazy_tensor._cholesky(upper=upper),
```

Comment:
```suggestion
            self.base_lazy_tensor._cholesky(upper=upper)._tensor,
```

Reply:
That has been causing issues if this is recursively called with `TriangularLazyTensor`s. What I've done instead is the constructor of `TriangularLazyTensor` strip out the underlying tensor if instantiated with `TriangularLazyTensor`.

Reply:
there may be a more elegant way to do this.